stdlib/LibGit2/src/repository.jl

```diff
@@ -140,9 +140,9 @@ function (::Type{T})(repo::GitRepo, spec::AbstractString) where T<:GitObject
     obj_ptr_ptr = Ref{Ptr{Cvoid}}(C_NULL)
     @assert repo.ptr != C_NULL
     @check ccall((:git_revparse_single, libgit2), Cint,
-                 (Ptr{Ptr{Cvoid}}, Ptr{Cvoid}, Cstring), obj_ptr_ptr, repo.ptr, spec)
+                 (Ptr{Ptr{Cvoid}}, Ptr{Cvoid}, Cstring), obj_ptr_ptr, repo, spec)
     # check object is of correct type
-    if T != GitObject && T != GitUnknownObject
+    if T != GitObject && T != GitUnknownObject # FIXME free result on error?
```

Comment:
Probably a good idea for the `throw` path.

Reply:
Ok but this requires to call a function that depends on the object type. This is fixed in the updated PR.